decoder_test.go

```diff
@@ -1,6 +1,7 @@
 package struct2
 
 import (
+	"reflect"
 	"testing"
 )
 
```

Comment:
Test_value2StructValue?
I usually don't test the private method

Reply:
Yes, I wasn't sure it is working so I just added that test case.